devito/types/dense.py

```diff
@@ -958,7 +959,25 @@ def __indices_setup__(cls, **kwargs):
                 raise TypeError("Need either `grid` or `dimensions`")
         elif dimensions is None:
             dimensions = grid.dimensions
-        return dimensions
+
+        # Staggered indices
+        staggered = kwargs.get("staggered", None)
+        if staggered is CELL:
+            staggered_indices = dimensions
+        elif staggered is NODE:
+            staggered_indices = dimensions
+        else:
+            as_dict = {d: d for d in dimensions}
+            for s in as_tuple(staggered):
+                c, s = s.as_coeff_Mul()
+                as_dict.update({s: s + c * s.spacing/2})
+
+            staggered_indices = tuple(as_dict.values())
+        return tuple(dimensions), staggered_indices
```

Comment:
here you mean that even an `x+1` is a "staggered index", even though one isn't really using a staggered grid?

Reply:
what?